CODINGSTYLE.md

```diff
@@ -0,0 +1,443 @@
+# Coding Style Guide
+
+The Scala.js project has a strict policy regarding coding style.
+This is one of the cornerstones that has allowed Scala.js to maintain a clean, consistent and maintainable codebase.
+
+This document tries to document the style in use as much as possible to make it easier for everyone to contribute.
+
+A *few* of these rules are checked automatically using Scalastyle, but most of them are too complex to teach to an automated tool.
+
+Our coding style is driven and "maintained" by [@sjrd](https://github.com/sjrd), the original author of Scala.js.
+Please follow his directions if they are in conflict with the present document.
+Feel free to point out the conflict, though: we are always looking to improve this document.
```

Comment:
@gzm0 WDYT? I'm afraid I'm too involved to reason about this without personal interference on my own reasoning. ^^ (if you know what confirmation bias is, well, I'm in the middle of it, and even though I identified it, it seems I can't get out ...)